client_plugin/drivers/shared/dockerops/dockerops.go

```diff
@@ -142,13 +175,165 @@ func (d *DockerOps) VolumeRemove(volName string) error {
 }
 
 // StartSMBServer - Start SMB server
-func (d *DockerOps) StartSMBServer(volName string) bool {
-	log.Errorf("startSMBServer to be implemented")
-	return true
+func (d *DockerOps) StartSMBServer(volName string) (int, string, bool) {
+	var service swarm.ServiceSpec
+	var options dockerTypes.ServiceCreateOptions
+
+	// Name of the service
+	service.Name = volName + serviceNamePostfix
```

Comment:
A Samba service that is running without a volume is called an Orphan samba service.
This can be a result of an improper cleanup during unmount. I was hitting this now because as of now we don't have unmount.

Once we have an orphan service stuck for a volume, we cannot remount that volume as mount() will try to launch service with the same name. The name will conflict.